src/controllers/auth.controller.js

```diff
@@ -0,0 +1,268 @@
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+
+require('dotenv').config();
+
+const emailService = require('../services/email.service');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { ApiError } = require('../utils/api.error');
+const { User } = require('../models/user.model');
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuid.v4();
+
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!name || !email || !password) {
+    throw ApiError.badRequest('Not all credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (user) {
+    throw ApiError.badRequest('The user with this email already exists!');
+  }
+
+  const hashedPassword = await bcrypt.hash(
+    password,
+    parseInt(process.env.HASH_ROUNDS, 10),
+  );
+  const newUser = { name, email, password: hashedPassword };
+
+  emailService.sendVerificationEmail(email, activationToken);
+  userService.create({ ...newUser, activationToken });
+
+  res.sendStatus(201);
+};
+
+const sendAuthentication = async (res, user) => {
+  const normalizedUser = userService.normalize(user);
+  const refreshToken = jwtService.generateRefreshToken(normalizedUser);
+  const accessToken = jwtService.generateAccessToken(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshToken);
+
+  await res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    httpOnly: true,
+    sameSite: 'none',
+    secure: true,
+  });
+
+  res.send({ user: normalizedUser, accessToken });
+};
+
+const activate = async (req, res) => {
+  const { token } = req.params;
+  const { refreshToken } = req.cookies;
+
+  if (refreshToken) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  const user = await User.findOne({ where: { activationToken: token } });
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  const normalizedUser = userService.normalize(user);
+
+  await sendAuthentication(res, normalizedUser);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
```

Comment:
The error message 'Only for not authorized' could be improved for clarity. Consider rephrasing it to something like 'Action restricted to unauthorized users only.'